### PR TITLE
version with deployment

### PR DESCRIPTION
**Pull request by hkollmann**
can be triggered wih

curl \
     -X POST \
     -H "Authorization: token <your_token>" \
     -H "Accept:application/vnd.github.everest-preview+json" \
     -H "Content-Type:application/json" \
     -d '{"event_type":"deploy"}' \
     https://api.github.com/repos/qooxdoo/deployment/dispatches

### CODE REVIEW

**Comment by hkollmann** [comment]
This will reflect the old state of publishing beta versions with timestamp in the name.
Semantic releasing based on tag revisions is still open.

But we need the old release state urgently.

File: deployer.sh [hunk 1]
```diff
@@ -117,14 +117,13 @@ if [[ $VERBOSE != 0 ]] ; then
...
     echo "USAGE=$USAGE"
-fi
-
 
-if [[ ! -d ./node_modules || ! -x $KNOWN_GOOD_QX_CMD ]]; then
-    info "Running npm install in deployment"
-    npm install
+    echo node version: $(node --version)
+    echo npm version:  $(npm --version)
+    echo bootstrap qx: $KNOWN_GOOD_QX_CMD
```
Comment (cboulanger) [inline deployer.sh]: Since the variable will always be "npx qx", wouldn't it be more interesting to display the version `$($KNOWN_GOOD_QX_CMD --version)` of the qx command?

File: deployer.sh [hunk 2]
```diff
@@ -272,14 +270,86 @@ if [[ $RUN_TESTS != 0 ]] ; then
...
+PACKAGE_DATE=$(date +%Y%m%d-%H%M)
+[[ ! -d $WORKING_ABS_DIR/deploy ]] && mkdir -p $WORKING_ABS_DIR/deploy
+rm -fR $WORKING_ABS_DIR/deploy/*
 
+function publishFramework {
+    info "publish qooxdoo framework"
+
+    pushDirSafe ${REPO_ABS_DIRS["qooxdoo"]}
+    local VERSION=$(cat version.txt)
+    VERSION="$VERSION-$PACKAGE_DATE"
```
Comment (cboulanger) [inline deployer.sh]: When v6 is released, the package date suffix needs to go. We have time until then, but maybe you want to make it conditional already now, maybe dependent on whether "alpha" or "beta" is part of the version? Then the suffix would be added automatically as soon as a alpha or beta version is being published.

Comment (hkollmann) [reply deployer.sh]: How can be this done in bash?

Comment (cboulanger) [reply deployer.sh]:
```bash
if [[ "$VERSION" =~ (alpha|beta) ]]; then
  VERSION="$VERSION-$PACKAGE_DATE"
fi
```

Comment (cboulanger) [reply deployer.sh]: The question is if we should rely on `version.txt` at all as the authoritative version info. But we can solve that later...

**Comment by hkollmann** [comment]
results are found here: https://github.com/hkollmann/deployment/actions

**Review by cboulanger — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

File: deployer.sh [hunk 3]
```diff
@@ -44,22 +42,6 @@ while [[ $1 != "" ]] ; do
             shift
             ;;
 
-        "--pull-all"|"-a")
```
Comment (cboulanger) [inline deployer.sh:47]: why did you remove those options? were they not useful?

Comment (hkollmann) [reply deployer.sh:47]: It's not used anywhere. I removed all params that are not used yet. If we use it we can put it in again.
